src/darwin/Framework/CHIP/MTRKeypair.h

```diff
@@ -31,13 +32,19 @@ NS_ASSUME_NONNULL_BEGIN
  * framework APIs.
  */
 @protocol MTRKeypair <NSObject>
-@required
+
+@optional
 /**
- * @brief Return public key for the keypair.
+ * @brief Returns a copy of the public key for the keypair.
  */
-- (SecKeyRef)publicKey;
+- (SecKeyRef)copyPublicKey MTR_NEWLY_AVAILABLE;
+
+/**
+ * @brief Returns public key for the keypair without adding a reference. DEPRECATED - please use copyPublicKey, otherwise this will leak.
+ */
+
+- (SecKeyRef)publicKey MTR_DEPRECATED("Please implement copyPublicKey, this will leak otherwise", ios(16.1, 18.3), macos(13.0, 15.3), watchos(9.1, 11.3), tvos(16.1, 18.3));
```

Comment:
Probably need 18.4/etc here, but we need to sort that out.  Might be simpler to do `MTR_NEWLY_DEPRECATED` so we handle it properly when we deal with `MTR_NEWLY` bits.

Reply:
changed to `MTR_NEWLY_DEPRECATED`.